plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/LdapGroupProvider.java

```diff
@@ -0,0 +1,198 @@
+                        this.bindDistinguishedName,
+                        this.bindPassword);
+                String groupFilter = replaceUser(this.groupSearchFilter, userDistinguishedName);
+                log.debug("Querying LDAP from %s using filter %s", this.groupBaseDistinguishedName, groupFilter);
+                Set<String> queryResult = lookupUserDistinguishedNames(
+                        this.groupBaseDistinguishedName, groupFilter,
+                        bindDistinguishedName, bindPassword);
+                groupSet = queryResult.stream().map(source -> {
+                    try {
+                        log.debug("extracting group from %s", source);
+                        Matcher m = EXTRACT_GROUP_PATTERN.matcher(source);
+                        if (m.find()) {
+                            return m.group(1);
+                        }
+                        log.debug("Cannot extract group cn [%s]", source);
+                        return "";
+                    }
+                    catch (Exception e) {
+                        log.debug(e, "Unknown exception, %s", e.getMessage());
+                        return "";
+                    }
+                }).filter(Objects::nonNull).collect(Collectors.toSet());
+            }
+            log.debug("Getting groups successful for user [%s]", user);
+            log.debug("Groups: %s", groupSet.toString());
+        }
+        catch (AccessDeniedException e) {
+            if (allowUserNotExist) {
+                return Collections.emptySet();
+            }
+            else {
+                log.debug(e, "Getting user failed [%s], %s", user, e.getMessage());
+                throw e;
+            }
+        }
+        catch (NamingException e) {
+            log.debug(e, "Getting groups failed for user [%s], %s", user, e.getMessage());
+            throw new RuntimeException(e);
+        }
+        return groupSet;
+    }
+
+    public Set<String> lookupUserDistinguishedNames(String searchBase, String searchFilter,
+            String contextUserDistinguishedName, String contextPassword)
+            throws NamingException
+    {
+        return client.executeLdapQuery(
+                contextUserDistinguishedName,
+                contextPassword,
+                new LdapQuery.LdapQueryBuilder()
+                        .withSearchBase(searchBase)
+                        .withSearchFilter(searchFilter)
+                        .build(),
+                searchResults -> {
+                    ImmutableSet.Builder<String> distinguishedNames = ImmutableSet.builder();
+                    while (searchResults.hasMore()) {
+                        distinguishedNames.add(searchResults.next().getNameInNamespace());
+                    }
+                    return distinguishedNames.build();
+                });
+    }
+
+    String lookupUserDistinguishedName(String userBaseDistinguishedName, String groupAuthorizationSearchPattern,
```

Comment:
Can we move each argument to a new line and provide a access sepcifier to this

Reply:
yup, fixing.